modules/cli/src/main/scala/scala/cli/commands/scalafix/Scalafix.scala

```diff
@@ -0,0 +1,174 @@
+      )
+    )
+    val inputs        = options.shared.inputs(args.all).orExit(logger)
+    val threads       = BuildThreads.create()
+    val compilerMaker = options.shared.compilerMaker(threads).orExit(logger)
+    val configDb      = ConfigDbUtils.configDb.orExit(logger)
+    val actionableDiagnostics =
+      options.shared.logging.verbosityOptions.actions.orElse(
+        configDb.get(Keys.actions).getOrElse(None)
+      )
+
+    val (sourcePaths, workspace, _) =
+      if (args.all.isEmpty)
+        (Seq(os.pwd), os.pwd, None)
+      else {
+        val s = inputs.sourceFiles().collect {
+          case sc: Script          => sc.path
+          case sc: SourceScalaFile => sc.path
+        }
+        (s, inputs.workspace, Some(inputs))
+      }
+
+    val scalaVersion =
+      options.buildOptions.orExit(logger).scalaParams.orExit(logger).map(_.scalaVersion)
+        .getOrElse(Constants.defaultScalaVersion)
+    val configFilePathOpt = options.scalafixConf.map(os.Path(_, os.pwd))
+    val relPaths          = sourcePaths.map(_.toNIO.getFileName)
+    val toolClasspath     = options.shared.dependencies.compileOnlyDependency
+
+    val scalafix = ScalafixInterface
+      .fetchAndClassloadInstance("2.13")
+      .newArguments()
+      .withParsedArguments(options.scalafixArg.asJava)
+      .withWorkingDirectory(workspace.toNIO)
+      .withPaths(relPaths.asJava)
+      .withRules(options.rules.asJava)
+      .withConfig(Optional.ofNullable(configFilePathOpt.map(_.toNIO).orNull))
+      .withScalaVersion(scalaVersion)
+      .withToolClasspath(Seq.empty.asJava, toolClasspath.asJava)
+      .withScalacOptions(Seq("-Wunused", "-Wunused:imports", "-P:semanticdb:synthetics:on").asJava)
+
+    val rulesThatWillRun = scalafix.rulesThatWillRun().asScala
+
+    logger.debug(
+      s"Rewriting and linting ${sourcePaths.size} Scala sources against ${rulesThatWillRun.size} rules"
+    )
+
+    val isSemantic = rulesThatWillRun.exists(_.kind().isSemantic)
+
+    val preparedScalafixInstance = if (isSemantic || toolClasspath.nonEmpty) {
+      val res = Build.build(
+        inputs,
+        buildOptionsWithSemanticDb,
+        compilerMaker,
+        None,
+        logger,
+        crossBuilds = false,
+        buildTests = false,
+        partial = None,
+        actionableDiagnostics = actionableDiagnostics
+      )
+      val builds = res.orExit(logger)
+
+      val successfulBuildOpt = for {
+        build <- builds.get(Scope.Main)
+        sOpt  <- build.successfulOpt
+      } yield sOpt
+
+      val classPaths = successfulBuildOpt.map(_.fullClassPath).getOrElse(Seq.empty)
+      val externalDeps =
+        options.shared.dependencies.compileOnlyDependency ++ successfulBuildOpt.map(
+          _.options.classPathOptions.extraCompileOnlyJars
+        ).getOrElse(Seq.empty).map(_.toNIO.toString)
+
+      scalafix
+        .withClasspath(classPaths.map(_.toNIO).asJava)
+        .withToolClasspath(Seq.empty.asJava, externalDeps.asJava)
```

Comment:
Same comment as above regarding toolClasspath, I am not sure what's the intent here?